### ISSUE TITLE
Alternative weighting scheme for Hessian elements based on Hagler

### ISSUE THREAD

**Issue by ericchansen** [Q2MM/q2mm#68]
Add new weighting scheme for Hessian elements.

Hessian elements always connect two atoms. The new scheme should allow weights to be assigned to elements based on the number of atoms between the pair of atoms corresponding to the Hessian element. 

- If it’s the same atom, this corresponds to an atom interacting with itself. It's a block diagonal element. We can call this weight `h11`. This weight will typically be set to zero.
- If the atoms are directly bonded, the Hessian elements define a bond force constant, so the weight should be high. We can call this weight `h12`.
- If they are separated by two bonds, that’s an angle, so it could be lower. We can call this weight `h13`.
- Three bonds means a torsion. Here we should follow Hagler and used a much higher weight to allow the torsions to be fit from Hessian, not from equilibrium structure. We can call this weight `h14`.
- More than three bonds mean it’s a non-bonded interaction, so unless we’re letting Hessian elements influence charges, these could be set to zero. We can call this weight `h15+`.

**Comment by ericchansen** [comment]
I see that some of the name ideas are already taken. Please refer to the snippet below.

https://github.com/Q2MM/q2mm/blob/71247cd4e7ba644edbe9f55eacd5c88da2e77183/q2mm/constants.py#L93-L115

I'm not aware of what `h12`, `h13` and `h14` mean in this context.

Is the option described by this issue already available?

**Comment by v3op01** [comment]
I added these terms specifically for Amber Hessian Matrix. It's in calculate.py

**Comment by ericchansen** [comment]
We can reuse them then, right? Since they refer to the same data type?

**Comment by v3op01** [comment]
Yes, I am missing h11, which I manually zeroed it in calculate.py and have h15 term as 0.031 (h term)

**Comment by ericchansen** [comment]
Okay. Sounds like this enhancement request still makes sense and should work for Amber and MacroModel.

**Comment by v3op01** [comment]
it should. For the part I added, I only made it so that it works only with Amber

**Comment by peonor** [comment]
A couple of comments: we need the scheme for MacroModel also (and presumably in Tinker when we start on it) so it should not be limited to Amber only. I'm not sure how to activate this weighting, is it the "-w" option in calculate.py? If so, I don't see how to change specific weights, "-w" doesn't take arguments.
I can find h12, h13, and h14 in constants.py, but I don't know how to activate them for MacroModel, and I'll also need the h11 set to zero (which should be default behavior when doing cartesian Hessian optimization). I don't see the need for h15+; let's see the numbered h's as specific replacements, and the "h" as the general, to be used for all interactions that are not special cases. I don't think the long range Hessian elements should be zeroed; either you have no parameters influencing them, and then they are a constant error, which doesn't affect a minimization, or you actually include the Hessian with nonbonded parameters, in which case you definitely need them.
